src/oauth2-client/index.ts

```diff
@@ -200,6 +201,23 @@ abstract class OAuth2Client {
   ): Promise<Response> {
     const { serverConfig } = Config.get(options);
     const url = this.getUrl(endpoint, query, options);
+    let actionType: string;
```

Comment:
It might be nice to use this pattern instead.

``` typescript
enum ActionType {
    'AUTHORIZE',
    'LOGOUT',
    'REVOKE_TOKEN',
     'USER_INFO'
}

type Actions = 'accessToken' | 'endSession' | 'revoke';

const actionType = (endpoint: Actions): keyof ActionType => {
    switch (endpoint) {
      case 'accessToken':
        return 'AUTHORIZE';
      case 'endSession':
        return 'LOGOUT';
      case 'revoke':
        return 'REVOKE_TOKEN';
      default:
        return 'USER_INFO';
    }
} 

const valueOfActionType = actionType(endpoint);
```


this way you can retain immutability.

Reply:
I like this idea. I'll make this change.

Reply:
I do love a switch statement return, its like the Rust lang match statement.